components/scripts/gradle/gradle-init-scripts/configure-remote-build-caching.gradle

```diff
@@ -9,11 +9,24 @@ if (isTopLevelBuild) {
             remote(HttpBuildCache) {
                 enabled = true
                 push = false
-                def projectProperties = gradle.startParameter.projectProperties
-                if (projectProperties.containsKey("com.gradle.enterprise.build_validation.remoteBuildCacheUrl")) {
-                    url = projectProperties.get("com.gradle.enterprise.build_validation.remoteBuildCacheUrl")
+                def remoteBuildCacheUrl = getRemoteBuildCacheUrl()
+                if (remoteBuildCacheUrl) {
+                    if (remoteBuildCacheUrl.path.endsWith("/")) {
+                        url = remoteBuildCacheUrl
+                    } else {
+                        url = withPathTrailingSlash(remoteBuildCacheUrl)
+                    }
                 }
             }
         }
     }
 }
+
+URI getRemoteBuildCacheUrl() {
+    def projectProperties = gradle.startParameter.projectProperties
+    new URI(projectProperties.get("com.gradle.enterprise.build_validation.remoteBuildCacheUrl"))
```

Comment:
I believe this will generate an NPE if the project property is missing.  And it looks like we checked to see if the project property existed before reconfiguring build caching.

So we may want to update this to check to see if the project property exists, and if not, return a placeholder (maybe null), so that the if statement on line 13 above will do what is intended.

Reply:
Nice catch. Addressed by 8d5237eb3a7924c296757edf233ed0d46b013e77